src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -146,6 +150,23 @@ private async Task ExecuteRequests(Hl7.Fhir.Model.Bundle responseBundle, Hl7.Fhi
                 if (request.Handler != null)
                 {
                     HttpContext httpContext = request.HttpContext;
+
+                    IFhirRequestContext originalFhirRequestContext = _fhirRequestContextAccessor.FhirRequestContext;
+
+                    request.RouteData.Values.TryGetValue(KnownActionParameterNames.ResourceType, out object resourceType);
+                    var newFhirRequestContext = new FhirRequestContext(
```

Comment:
shoudl we add a test to make sure that correct audit log is produced for the batch call? It might only run locally (since the existing audit tests only works with in-memory server at the moment) but would still be nice to have coverage?

Reply:
Yes, we should. The plan is to do that as part of the stories AB#70787 and AB#70788 once we change the feature flag to be enabled by default.